spec/requests/trains_request_spec.rb

```diff
@@ -24,6 +25,17 @@
     end
   end
 
+  describe "GET /create" do
+    before do
+      expect_any_instance_of(TrainApi).to receive(:new).and_return({})
+    end
```

Comment:
i dont think this is right.
i think you havent stubbed the right action and are unintentionally posting data to your api again.
as we dont call `TrainApi.new.new` we call `TrainApi.new.create`